src/Compilers/Core/Portable/Syntax/SyntaxTree.cs

```diff
@@ -102,7 +102,7 @@ public ParseOptions Options
         /// <summary>
         /// The text encoding of the source document.
         /// </summary>
-        public abstract Encoding Encoding { get; }
+        public abstract Encoding? Encoding { get; }
```

Comment:
When can this be null? It's certainly not something I would have expected to ever be null.

Reply:
I was surprised too, but the CSharp implementation is pretty explicit. It can happend when parsing a null `SourceText` or a `SourceText` with no encoding.

```
    public partial class CSharpSyntaxTree
    {
        private class ParsedSyntaxTree : CSharpSyntaxTree
        {
...
            private readonly Encoding? _encodingOpt;
...

            internal ParsedSyntaxTree(
                SourceText? textOpt,
                Encoding? encodingOpt,
...
                )
            {
...
                Debug.Assert(textOpt == null || textOpt.Encoding == encodingOpt && textOpt.ChecksumAlgorithm == checksumAlgorithm);
...
                _encodingOpt = encodingOpt ?? textOpt?.Encoding;
```

```
        public static SourceText From(string text, Encoding? encoding = null, SourceHashAlgorithm checksumAlgorithm = SourceHashAlgorithm.Sha1)
        {
...
            return new StringText(text, encoding, checksumAlgorithm: checksumAlgorithm);
        }
```